plugins/hfp_ag_bluez5.c

```diff
@@ -95,7 +109,10 @@ static DBusMessage *profile_new_connection(DBusConnection *conn,
 		goto invalid;
 
 	dbus_message_iter_get_basic(&entry, &fd);
+
 	dbus_message_iter_next(&entry);
+	if (dbus_message_iter_get_arg_type(&entry) != DBUS_TYPE_ARRAY)
```

Comment:
There is a mismatch between NewConnection() description in handsfree-audio-api.txt and the actual declaration in profile_methods[]. The first one says the third argument is a byte and not an array. Would be good to check with upstream which one is right. Maybe it is still not decided what the third argument should be, and it is not used for the moment anyway, so I would remove this check. Also, the fd should be closed if the check fails.